packages/payload/src/utilities/configToJSONSchema.ts

```diff
@@ -60,6 +60,16 @@ function generateEntitySchemas(
   }
 }
 
+function withNullableType(
```

Comment:
this `withNullableType` function is missing some checks that our other `withNullableType` found here https://github.com/payloadcms/payload/blob/main/packages/payload/src/graphql/schema/withNullableType.ts is checking for: 
conditions, localized, access control.

Shouldn't they be the same conditions?

Reply:
ooooo good catch, I can make this match that.

Reply:
Hmm actually looking at this closer, I think we should leave it as is. If I change it to match the GraphQL one, we will generate much different types than we currently are. Take the `read` access control check in the GraphQL one for example, if we used that then a ton of fields that before were required will now be optional. 

I think it would be better to leave them as is for now and address this properly in the future by generating specific types for CRUD operations.

Reply:
Yeah that makes sense with `read`. I'm not sure I agree for conditions though. Condition is not operation specific.

Reply:
I made the changes in [this function](https://github.com/payloadcms/payload/pull/3870/files#diff-50cbef23ef608a8e1c9bad6cbf9b65822b206d2b4d618d6ff89455381c67ca06R14-R37)

I removed the read check from fields with fields because it felt odd that we were only targeting group fields with that condition, and again I think we should just improve types in the future.